FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -72,6 +74,12 @@ public class ModelDownloader {
     appName = app.name
     options = app.options
     installations = Installations.installations(app: app)
+    /// Respect Firebase-wide data collection setting.
+    telemetryLogger = TelemetryLogger(
+      isStatsEnabled: app.isDataCollectionDefaultEnabled,
```

Comment:
`isDataCollectionDefaultEnabled` is a runtime flag that could be changed at any point. It's likely better to pass the app to `TelemetryLogger` and have it read from the flag each time it's about to send data. Otherwise, a user could opt out of logging and we'd still be sending things.

Reply:
Great catch! Updated.